functions.php

```diff
@@ -84,3 +84,65 @@ function twentytwentyfour_block_styles() {
 endif;
 
 add_action( 'init', 'twentytwentyfour_block_styles' );
+
+/**
+ * Register a block variation for a query loop with only featured images.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/
+ * @see https://developer.wordpress.org/news/2022/12/building-a-book-review-grid-with-a-query-loop-block-variation/
+ * @return void
+ * @since Twenty Twenty-Four 1.0
+ *
+ */
+function twentytwentyfour_register_block_variation() {
+	wp_enqueue_script(
+		'twentytwentyfour-block-variations',
+		get_template_directory_uri() . '/assets/js/block-variation.js',
+		array( 'wp-blocks', 'wp-i18n' ),
+		wp_get_theme()->get( 'Version' ),
+		true
+	);
+}
+add_action( 'enqueue_block_assets', 'twentytwentyfour_register_block_variation' );
+
+function twentytwentyfour_rest_filter_query( $args, $request ) {
+	$has_featured_image = $request->get_param( 'hasFeaturedImage' );
+	if ( $has_featured_image ) {
```

Comment:
The custom `hasFeaturedImage` doesn't seem to exist, for example it is not included if log the content of `get_params()`.
So the condition is never true and the result includes posts without featured images.